contracts/incentives/base/BaseIncentivesController.sol

```diff
@@ -201,30 +250,35 @@ abstract contract BaseIncentivesController is
     if (amount == 0) {
       return 0;
     }
-    uint256 unclaimedRewards = _usersUnclaimedRewards[user];
+    uint256 totalUnclaimedRewards = _usersUnclaimedRewards[user];
 
-    if (amount > unclaimedRewards) {
-      DistributionTypes.UserStakeInput[] memory userState =
-        new DistributionTypes.UserStakeInput[](assets.length);
-      for (uint256 i = 0; i < assets.length; i++) {
-        userState[i].underlyingAsset = assets[i];
-        (userState[i].stakedByUser, userState[i].totalStaked) = IScaledBalanceToken(assets[i])
-          .getScaledUserBalanceAndSupply(user);
-      }
-
-      uint256 accruedRewards = _claimRewards(user, userState);
-      if (accruedRewards != 0) {
-        unclaimedRewards = unclaimedRewards.add(accruedRewards);
-        emit RewardsAccrued(user, accruedRewards);
-      }
+    DistributionTypes.UserStakeInput[] memory userState =
+      new DistributionTypes.UserStakeInput[](assets.length);
+    for (uint256 i = 0; i < assets.length; i++) {
+      userState[i].underlyingAsset = assets[i];
+      (userState[i].stakedByUser, userState[i].totalStaked) = IScaledBalanceToken(assets[i])
+        .getScaledUserBalanceAndSupply(user);
     }
 
-    if (unclaimedRewards == 0) {
+    uint256 accruedRewards = _claimRewards(user, userState);
+    if (accruedRewards != 0) {
+      totalUnclaimedRewards = totalUnclaimedRewards.add(accruedRewards);
+      emit RewardsAccrued(user, accruedRewards);
+    }
+
+    if (totalUnclaimedRewards == 0) {
       return 0;
     }
 
-    uint256 amountToClaim = amount > unclaimedRewards ? unclaimedRewards : amount;
-    _usersUnclaimedRewards[user] = unclaimedRewards - amountToClaim; // Safe due to the previous line
+    // uint256 amountToClaim = amount > unclaimedRewards ? unclaimedRewards : amount;
+    // _usersUnclaimedRewards[user] = unclaimedRewards - amountToClaim; // Safe due to the previous line
+    uint256 userTotalRewards = totalUnclaimedRewards.add(_userClaimedRewards[user]);
+    uint256 amountToClaim = distributedAmount(userTotalRewards).sub(_userClaimedRewards[user]);
```

Comment:
This seems incompatible with the original interface because the function parameter `amount` is not taken into consideration. Is this OK?

Reply:
yeah the interface is still compatible, but I changed the semantics from claiming certain amount of rewards to claim all for the simplicity of implementation.

I will make sure the UI adopts the change as well.

Reply:
the interface is still compatible, but I changed the semantics from claiming certain amount of rewards to claim all for the simplicity of implementation. I will make sure the UI adopts the change as well.

Reply:
sounds good

Reply:
Why not delete `amount`? Is it because of `IAaveIncentivesController `?

Reply:
yep i want to keep the interface so that there won't be a lot of changes from the client side